source/adios2/toolkit/derived/parser/ASTNode.cpp

```diff
@@ -76,6 +76,43 @@ std::string ASTNode::printpretty(std::string indent)
     return result;
 }
 
+std::string ASTNode::node_to_string_expr()
```

Comment:
Since we have a way of printing the Expression string, do we need this at the `ASTNode` level as well?

Reply:
No you're right, we can replace the ASTNode::node_to_string_expr function with the ExpressionTree function. I've removed the ASTNode function